### PR TITLE
Add database integration related templates and guides

### LINKED ISSUE #332: Add database integration related templates and migrate the existing guides
**Description:**
Add the following database integration-related templates: 
> data-backed services
> managing database transactions

Migrate the existing guides for the same to the latest Ballerina version (1.0.0).

**Affected Product Version:**
Ballerina Integrator

### PR DESCRIPTION
**Pull request by IrushiL**
## Purpose
* Add data_backed_service template.
* Add managing_database_transactions template
* Migrate data_backed_service guide to the latest Ballerina version (1.0.0).
* Migrate managing_database_transactions guide to the latest Ballerina version (1.0.0).

resolves #332

### CODE REVIEW

File: templates/database-integrations/data_backed_service/src/data_backed_service/employee_db_service.bal [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+        if (payloadJson is json) {
+            Employee | error employeeData = Employee.constructFrom(payloadJson);
+
+            if (employeeData is Employee) {
+                // Validate JSON payload
+                if (employeeData.name == "" || employeeData.age == 0 || employeeData.ssn == 0 ||
+                employeeData.employeeId == 0) {
+                    response.setPayload("Error : json payload should contain
+                        {name:<string>, age:<int>, ssn:<123456>, employeeId:<int>}");
+                    response.statusCode = 400;
```
Comment (Asitha) [inline templates/database-integrations/data_backed_service/src/data_backed_service/employee_db_service.bal]: Shall  we use constants?

File: templates/database-integrations/data_backed_service/src/data_backed_service/employee_db_service.bal [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+        // Convert the sql data table into JSON using type conversion
+        jsonReturnValue = jsonutils:fromTable(ret);
+    } else {
+        jsonReturnValue = {"Status": "Data Not Found", "Error": "Error occurred in data retrieval"};
+        log:printError("Error occurred in data retrieval", err = ret);
+    }
+    return jsonReturnValue;
+}
+
+public function updateData(string name, int age, int ssn, int employeeId) returns (json) {
```
Comment (Asitha) [inline templates/database-integrations/data_backed_service/src/data_backed_service/employee_db_service.bal]:
```suggestion
public function updateData(string name, int age, int ssn, int employeeId) returns json {
```

File: templates/database-integrations/data_backed_service/src/data_backed_service/Module.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
+# Data Backed Service
+
+Data inside a database can be exposed to the outside world by using a database backed RESTful web service. RESTful API calls enable you to add, view, update, and remove data stored in a database from the outside world. This template demonstrates on building a database-backed RESTful web service with Ballerina.
```
Comment (samgnaniah) [inline templates/database-integrations/data_backed_service/src/data_backed_service/Module.md]:
```suggestion
Data inside a database can be exposed to the outside world by using a database backed RESTful web service. RESTful API calls enable you to add, view, update, and remove data stored in a database from the outside world. This template demonstrates how to build a database-backed RESTful web service with Ballerina.
```

File: templates/database-integrations/data_backed_service/src/data_backed_service/Module.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+Please use the guide documentation [Database Interaction](https://github.com/wso2/ballerina-integrator/tree/master/docs/learn/guides/database-integrations/data-backed-service) for a more detailed explanation.
```
Comment (samgnaniah) [inline templates/database-integrations/data_backed_service/src/data_backed_service/Module.md]: Better use relative links here. When this is rendered on the doc site it has to point to a doc page, not GitHub.

File: templates/database-integrations/data_backed_service/src/data_backed_service/Module.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+## Compatibility
+| Ballerina Language Version  | 
+|:---------------------------:|
+|  1.0.0                     |
+
+## Prerequisites
```
Comment (samgnaniah) [inline templates/database-integrations/data_backed_service/src/data_backed_service/Module.md]: Let's add VS Code and Java as well.

File: templates/database-integrations/data_backed_service/src/data_backed_service/Module.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+| EmployeeID | int         | NO   | PRI | NULL    |       |
+| Name       | varchar(50) | YES  |     | NULL    |       |
+| Age        | int         | YES  |     | NULL    |       |
+| SSN        | int         | YES  |     | NULL    |       |
++------------+-------------+------+-----+---------+-------+
+
+```
+
+## Running the Template
+Configure the ballerina.conf file inside `src/<module_name>/resources` directory with relevant configuration.
```
Comment (samgnaniah) [inline templates/database-integrations/data_backed_service/src/data_backed_service/Module.md]:
```suggestion
Configure the `ballerina.conf` file inside `src/<module_name>/resources` directory with the relevant configuration.
```

File: templates/database-integrations/data_backed_service/src/data_backed_service/Module.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+Execute the following Ballerina command from within the Ballerina project folder to run the data backed service template module.
+```ballerina    
+ballerina run <module_name> --b7a.config.file=<path_to_ballerina.conf_file>
```
Comment (samgnaniah) [inline templates/database-integrations/data_backed_service/src/data_backed_service/Module.md]: Shall we use the module name used in this guide?

File: templates/database-integrations/data_backed_service/src/data_backed_service/employee_db_service.bal (hunk 1 above)
Comment (IrushiL) [reply templates/database-integrations/data_backed_service/src/data_backed_service/employee_db_service.bal]: Done

File: templates/database-integrations/data_backed_service/src/data_backed_service/employee_db_service.bal (hunk 2 above)
Comment (IrushiL) [reply templates/database-integrations/data_backed_service/src/data_backed_service/employee_db_service.bal]: Done

File: docs/learn/guides/database-integrations/data-backed-service/README.md [hunk 8]
```diff
@@ -10,29 +11,25 @@ The following are the sections available in this guide.
...
 
-You'll build an employee data management REST service that performs CRUD Operations (Create, Read, Update, Delete) on the MySQL database. Also, this guide walks you through the process of accessing relational data via the Ballerina language. The service will have following functionalities.
+You'll build an employee data management REST service that performs CRUD Operations (Create, Read, Update, Delete) on the MySQL database. Also, this guide walks you through the process of accessing relational data via the Ballerina language. The service will have the following functionalities.
 
 * Add new employees to the database via HTTP POST method
-* Retrieve an existing employee details from the database via HTTP GET method
-* Update an existing employee in the database via HTTP PUT method
+* Retrieve details of an existing employee from the database via HTTP GET method
+* Update an existing employee in the database via the HTTP PUT method
 * Delete an existing employee from the database via HTTP DELETE method
```
Comment (samgnaniah) [inline docs/learn/guides/database-integrations/data-backed-service/README.md]:
```suggestion
* Delete an existing employee from the database via the HTTP DELETE method
```

File: docs/learn/guides/database-integrations/data-backed-service/README.md [hunk 9]
```diff
@@ -42,36 +39,37 @@ Basically, this service will deal with a MySQL database and expose the data oper
...
+1. Create a project.
+ ```bash
+ $ ballerina new data-backed-service
+ ```
 
-- Then open the terminal and navigate to `data-backed-service/guide` and run Ballerina project initializing toolkit.
-```bash
-   $ ballerina init
-```
+ 2. Move into the project directory and add a new module.
```
Comment (samgnaniah) [inline docs/learn/guides/database-integrations/data-backed-service/README.md]:
```suggestion
 2. Navigate to the project directory and add a new module.
```

File: docs/learn/guides/database-integrations/data-backed-service/README.md [hunk 10]
```diff
@@ -42,36 +39,37 @@ Basically, this service will deal with a MySQL database and expose the data oper
...
+ ```bash
+ $ ballerina add data_backed_service_module
+ ```
 
-### Developing the SQL data backed web service
+ 3. Add a ballerina.conf file and rename the bal files with meaningful names as shown in the project structure given below.
```
Comment (samgnaniah) [inline docs/learn/guides/database-integrations/data-backed-service/README.md]:
```suggestion
 3. Add a ballerina.conf file and rename the .bal files with meaningful names as shown in the project structure given below.
```

File: docs/learn/guides/database-integrations/data-backed-service/README.md [hunk 11]
```diff
@@ -83,261 +81,10 @@ We can use the following database schema to store employee data.
 | Age        | int(11)     | YES  |     | NULL    |       |
 | SSN        | int(11)     | YES  |     | NULL    |       |
 +------------+-------------+------+-----+---------+-------+
-
 ```
-The following Ballerina code is the employee data service with resource functions to add, retrieve, update and delete employee data.
+The Ballerina code for the employee data service with resource functions to add, retrieve, update and delete employee data can be found in [employee_db_service.bal](src/data_backed_service_module/employee_db_service.bal).
```
Comment (samgnaniah) [inline docs/learn/guides/database-integrations/data-backed-service/README.md]: Will this link work when we render it on the docs site?

File: docs/learn/guides/database-integrations/data-backed-service/README.md [hunk 12]
```diff
@@ -355,27 +102,32 @@ You can implement custom functions in Ballerina that perform specific tasks. For
...
 
-- To run the developed employee database service you need to navigate to `data-backed-service/guide` and execute the following command
+Let’s build the module. Navigate to the project directory and execute the following command.
+
+```bash
+$ ballerina build data_backed_service_module
 ```
-$ ballerina run data_backed_service
+
+The build command would create an executable jar file. Now run the jar file created in the above step.
```
Comment (samgnaniah) [inline docs/learn/guides/database-integrations/data-backed-service/README.md]:
```suggestion
The build command would create an executable .jar file. Now run the .jar file created in the above step.
```

File: docs/learn/guides/database-integrations/data-backed-service/README.md [hunk 13]
```diff
@@ -355,27 +102,32 @@ You can implement custom functions in Ballerina that perform specific tasks. For
...
+
+```bash
+$ java -jar target/bin/data_backed_service_module.jar
 ```
 
-- You can test the functionality of the employee database management RESTFul service by sending HTTP requests for each database operation. For example, this guide uses the cURL commands to test each operation of employeeService as follows. 
+- Now you can test the functionality of the employee database management RESTFul service by sending HTTP requests for each database operation. For example, this guide uses the cURL commands to test each operation of employeeService as follows.
```
Comment (samgnaniah) [inline docs/learn/guides/database-integrations/data-backed-service/README.md]:
```suggestion
- Now you can test the functionality of the employee database management RESTFul service by sending HTTP requests for each database operation. For example, this guide uses the cURL commands to test each operation of the `employeeService` as follows.
```

File: docs/learn/guides/database-integrations/data-backed-service/README.md [hunk 14]
```diff
@@ -409,498 +161,19 @@ Output:
...
-filebeat.prospectors:
-- type: log
-  paths:
-    - /usr/share/filebeat/ballerina.log
-output.logstash:
-  hosts: ["logstash:5044"]  
-```
-NOTE : Modify the ownership of filebeat.yml file using `$chmod go-w filebeat.yml` 
+To run the unit tests, go to the project directory and run the following command.
+Please note that `--b7a.config.file=path/to/file` option is required if it is needed to read configurations from a ballerina configuration file.
```
Comment (samgnaniah) [inline docs/learn/guides/database-integrations/data-backed-service/README.md]:
```suggestion
> **Note**: The `--b7a.config.file=path/to/file` option is required if it is needed to read configurations from a Ballerina configuration file.
```

**Comment by IrushiL** [comment]
Addressing this PR separately due to changes in the folder structure.

**Pull request closed without merging**